src/lib.rs

```diff
@@ -111,6 +114,16 @@ impl From<u32> for Pid {
     }
 }
 
+impl FromStr for Pid {
+    type Err = <u32 as FromStr>::Err;
```

Comment:
I think it would be fine to make this conversion return `blazesym`'s `Error` type. Then we could include the relevant context right away, instead of pushing that to all call sites:

```rust
impl FromStr for Pid {
    type Err = Error;

    fn from_str(s: &str) -> Result<Self, Self::Err> {
        s.parse::<u32>().map(Self::from).map_err(|_| {
            Error::with_invalid_data(format!("failed to parse input `{s}` as process ID"))
        })
    }
}
```